test/unit/lldp_neighbors_test.rb

```diff
@@ -0,0 +1,54 @@
+require 'test_plugin_helper'
+
+class LldpNeighborsTest < ActiveSupport::TestCase
+  test "#get_neighbors_by_interface gives nothing with no LLDP facts" do
+    assert_nil simple_facts.get_neighbors_by_interface('eth0')
+  end
+
+  test "#get_neighbors_by_interface gives nothing with unmatching PVID in LLDP facts" do
+    assert_nil lldp_facts.get_neighbors_by_interface('eth0')
+  end
+
+  test "#get_neighbors_by_interface gives a valid pair for LLDP facts with matching PVID" do
+    assert_equal %w(eth0 eth1), bond_facts.get_neighbors_by_interface('eth0')
+    assert_equal %w(eth0 eth1), bond_facts.get_neighbors_by_interface('eth1')
+    assert_nil bond_facts.get_neighbors_by_interface('br182')
+  end
+
+  test "#list_by_pvid gives an empty Hash without LLDP facts" do
+    assert_equal Hash({}), simple_facts.list_by_pvid
+    assert_equal Hash({}), ForemanDiscovery::LldpNeighbors.from_facts({}).list_by_pvid
+  end
+
+  test "#list_by_pvid gives an empty Hash without LLDP facts" do
+    assert_equal Hash('182' => %w(eth2), '184' => %w(eth1)), lldp_facts.list_by_pvid
+  end
+
+  test "#list_by_pvid pairs interfaces by PVID" do
+    assert_equal Hash('182' => %w(eth0 eth1)), bond_facts.list_by_pvid
+  end
+
+  test "#list_by_pvid supports multiple PVIDs" do
+    assert_equal Hash('182' => %w(eth2), '184' => %w(eth1)), lldp_facts.list_by_pvid
+  end
+
```

Comment:
the methods below can be private, a better (or modern) way to write this would be using `let`. But this is fine as well.

```ruby
context 'with facts' do
 let(:facts) { some code } # acts as a lazy accessor
  test 'abc' do
    assert facts
  end
end
```